cdc-accelerator/feature/acls.js

```diff
@@ -0,0 +1,59 @@
+import path from 'path'
+import fs from 'fs'
+import { clearDirectoryContents } from '../utils/utils.js'
+import { SRC_DIRECTORY, BUILD_DIRECTORY } from './constants.js'
+import client from '../../cdc-accelerator/sap-cdc-toolkit/gigya/client.js'
+export default class ACL {
+    static ACL_FILE_NAME = 'Acls.json'
+    #credentials
+    constructor(credentials) {
+        this.#credentials = credentials
+    }
+
+    getName() {
+        return this.constructor.name
+    }
+
+    async init(aclID, partnerId, permissionGroupDirectory, dataCenter) {
+        const promises = aclID.map(async (ids) => {
+            const response = await this.getAclsRequest(dataCenter, ids, partnerId, this.#credentials)
+            if (response.errorCode) {
+                throw new Error(JSON.stringify(response))
+            }
+
+            return { [ids]: response['acl'] }
+        })
+        const results = await Promise.all(promises)
+        let finalResponse = Object.assign({}, ...results)
+
+        fs.writeFileSync(path.join(permissionGroupDirectory, ACL.ACL_FILE_NAME), JSON.stringify(finalResponse, null, 4))
+    }
+
+    reset(directory) {
+        this.deleteDirectory(path.join(directory, this.getName()))
+    }
+
+    build(directory) {
+        const buildFeaturePath = path.join(directory, this.getName())
+        clearDirectoryContents(buildFeaturePath)
+        const srcFeaturePath = buildFeaturePath.replace(BUILD_DIRECTORY, SRC_DIRECTORY)
+        this.copyFileFromSrcToBuild(srcFeaturePath, ACL.ACL_FILE_NAME)
+    }
+
+    async deploy(partnerDirectory, siteInfo) {
+        console.log('deploy was called')
+    }
+    async getAclsRequest(dataCenter, requestBody, partnerID, credentials) {
```

Comment:
requestBody is a strange name for an acl id, consider renaming to increase the code readability